src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/playbackaudiovideo/PlaybackAudioVideoEvidenceMidEventHandler.java

```diff
@@ -0,0 +1,100 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.playbackaudiovideo;
+
+import static java.util.Objects.nonNull;
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.sscs.ccd.presubmit.playbackaudiovideo.PlaybackAudioVideoActionHelper.*;
+import static uk.gov.hmcts.reform.sscs.idam.UserRole.*;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.DocumentType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Component
+@Slf4j
+public class PlaybackAudioVideoEvidenceMidEventHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    private String dmGatewayUrl;
+    private String documentManagementUrl;
+
+    @Autowired
+    public PlaybackAudioVideoEvidenceMidEventHandler(@Value("${dm_gateway.url}") String dmGatewayUrl,
+                                                     @Value("${document_management.url}") String documentManagementUrl) {
+        this.dmGatewayUrl = dmGatewayUrl;
+        this.documentManagementUrl = documentManagementUrl;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.MID_EVENT)
+                && callback.getEvent() == EventType.PLAYBACK_AUDIO_VIDEO_EVIDENCE;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+
+
+        List<? extends AbstractDocument> sscsAvDocuments = getApprovedAudioVideoDocuments(sscsCaseData.getSscsDocument());
```

Comment:
You can move the code that gives you combinedAvDocuments to the helper class as you need it in both handlers